### PR TITLE
[sqlite3] Namespaced targets with unofficial::

### LINKED ISSUE #12505: [SQlite3] Missing unofficial:: namespace in generated target.
Since vcpkg owns the CMake build of sqlite3 and exports the targets via a CMakeLists.txt file in the port folder the target should be namespaced with `unoffical::` like all other targets vcpkg tends to export on its own.

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/12505

1. Namespaced sqlite3 targets with unofficial::
2. Update the ports(libgpod, libodb-sqlite, sqlitecpp) which depends on it.
3. All features test pass with x86-windows.

### CODE REVIEW

File: ports/sqlite3/CMakeLists.txt [hunk 1]
```diff
@@ -46,11 +46,11 @@ endif()
...
-    EXPORT sqlite3
+    EXPORT unofficial-sqlite3-targets
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
 )
 
 install(FILES sqlite3.h sqlite3ext.h DESTINATION include CONFIGURATIONS Release)
-install(EXPORT sqlite3 FILE sqlite3-targets.cmake DESTINATION share/sqlite3)
+install(EXPORT unofficial-sqlite3-targets NAMESPACE unofficial::sqlite3:: FILE sqlite3-targets.cmake DESTINATION share/unofficial-sqlite3)
```
Comment (Neumann-A) [inline ports/sqlite3/CMakeLists.txt:56]: is there a reason to double namespace the target? All other unofficial targets only have `unofficial::` instead `unofficial::<projectname>::` of  as a namespace

Comment (PhoebeHui) [reply ports/sqlite3/CMakeLists.txt:56]: I noticed that others ports(date, cairo, kissfft..) also add the projectname, so I added it for sqlite3 as well.

 I can remove the projectname here, should I update it?

Comment (Neumann-A) [reply ports/sqlite3/CMakeLists.txt:56]: Maybe you should internally discuss how you want to handle `unofficial` targets in a consistent way.

Comment (Neumann-A) [reply ports/sqlite3/CMakeLists.txt:56]: a wait forget that. I think `unofficial::sqlite3::` is ok just checked glib and libiconv which I thought did it differently but they also have the double namespace.

**Review by JackBoosY — APPROVED** [review]

**Comment by PhoebeHui** [comment]
We might need update related docs as well, I remember we has some example that use sqlite3.

**Comment by strega-nil** [comment]
@PhoebeHui could you look into that today and get back to me?

**Comment by PhoebeHui** [comment]
Sure.

**Comment by PhoebeHui** [comment]
@strega-nil, I have updated the example on github, could you help update the sqlite3 in doc https://vcpkg.readthedocs.io/en/latest/examples/installing-and-using-packages/

**Comment by strega-nil** [comment]
@PhoebeHui is there a reason that this is a draft PR?

**Comment by PhoebeHui** [comment]
@strega-nil, I usually mark it to 'ready to review' when the CI test pass.

**Comment by strega-nil** [comment]
@PhoebeHui ah, cool, thanks <3

**Comment by stoperro** [comment]
I think `dlib` port still references `sqlite3` not `unofficial-sqlite3` in: `installed\x64-windows\share\dlib\dlibConfig.cmake`:
`   find_dependency(sqlite3 CONFIG)`

**Comment by PhoebeHui** [comment]
@stoperro, thanks for reporting this issue, I sumiited PR https://github.com/microsoft/vcpkg/pull/13325 to update it.

**Pull request merged**